google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -213,13 +214,18 @@ public B setHost(String host) {
     }
 
     /**
-     * Sets the service authentication credentials.
+     * Sets the service authentication credentials. If this method or {@link #setNoCredentials() are
+     * not used on the builder, {@link GoogleCredentials#getApplicationDefault()} will be used to
+     * attempt getting credentials from the environment.
      *
+     * @param credentials authentication credentials, should not be {@code null}
      * @return the builder
+     * @throws NullPointerException if {@code credentials} is {@code null}. To disable
+     *     authentication use {@link Builder#setNoCredentials()}
```

Comment:
Given this, how does one take an existing options instance and make it use the default credentials?